### PR TITLE
fix: histogram with values zero & uniqueIdProperty from cartoLayerProps hook

### PR DESCRIPTION
**Pull request by AdriSolid**
For: https://app.clubhouse.io/cartoteam/story/141415/bug-thinking-machines-histogram
For: https://app.clubhouse.io/cartoteam/story/141411/uniqueidproperty-behaviour-in-hook

### CODE REVIEW

File: packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import { DataFilterExtension } from '@deck.gl/extensions';
 import { renderHook } from '@testing-library/react-hooks';
 import useCartoLayerProps from '../../src/hooks/useCartoLayerProps';
```
Comment (VictorVelarde) [inline packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js:3]: The name of this test file is OLD (LayerFilterProps...) and has to change

File: packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-import { mockClear, mockReduxHooks } from '../mockReduxHooks';
+import { mockReduxHooks, mockClear } from '../mockReduxHooks';
```
Comment (VictorVelarde) [inline packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js:4]: why this order change, automatic?

File: packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js (hunk 1 above)
Comment (AdriSolid) [reply packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js:3]: Good catch!

File: packages/react-api/reference.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ Returns required default props for layers. It manages filtering and viewport cha
 | Param                               | Type                          | Description                                                               |
 | ----------------------------------- | ----------------------------- | ------------------------------------------------------------------------- | ---------------------------------------------------- |
 | props                               | <code>Object</code>           | Required default props for layers                                         |
+| props.uniqueIdProperty              | <code>string</code>           | Unique id property for the layer, default to `cartodb_id`                 |
```
Comment (VictorVelarde) [inline packages/react-api/reference.md:28]: We must also add the uniqueIdProperty as output right?

File: packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js (hunk 2 above)
Comment (AdriSolid) [reply packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js:4]: Make more sense right? First, initialization, then clearing.

File: packages/react-api/reference.md (hunk 3 above)
Comment (AdriSolid) [reply packages/react-api/reference.md:28]: That block is actually the output

File: packages/react-core/src/operations/histogram.js [hunk 4]
```diff
@@ -17,7 +17,9 @@ export function histogram(features, columnName, ticks, operation) {
   features.forEach((feature) => {
     const featureValue = feature[columnName];
 
-    if (!featureValue) {
+    const isValid = featureValue !== null && featureValue !== undefined;
```
Comment (VictorVelarde) [inline packages/react-core/src/operations/histogram.js:20]: 🤦🏻 ... falsy value bites us again...

File: packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js (hunk 2 above)
Comment (VictorVelarde) [reply packages/react-api/__tests__/hooks/useCartoLayerFilterProps.test.js:4]: Ok, I tend to use alphabetic ordering, but it's ok. We should see if we can enforce a criteria in the future with linter

File: packages/react-api/reference.md (hunk 3 above)
Comment (VictorVelarde) [reply packages/react-api/reference.md:28]: sorry man, I didn't notice 👍🏻

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Pull request merged**